### PR TITLE
Refactor requestHandler and resolversFinder

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- some components needed renamings/to be moved/to use newly added containers
  
## Proposed changes
- moved `topicResolverSender` to `package topicsender`
- the storage resolvers are now storage requesters
- `resolversFinder` is now `requestersFinder`
- updated `requestHandler` to use `requestersFinder`
- many `*resolver*` -> `*requester*` renamings

## Testing procedure
- to be tested when feat branch is ready

### Note for review 
`requesterFinder.go` should be compared with the old `resolverFinder.go`, as git does not display it as moved but as a new file([diffChecker](https://www.diffchecker.com/yws9qTvR)). Same for the test file([diffChecker](https://www.diffchecker.com/hKbBVzKZ))

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/ElrondNetwork/elrond-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: dataRetriever/factory/containers/requestersFinder_test.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+			if key == expectedKey {
+				return &dataRetrieverMocks.RequesterStub{}, nil
+			}
+
+			return nil, nil
+		},
+	}
+}
+
+func TestNewRequestersFinder_NilContainerShouldErr(t *testing.T) {
```
Comment (ssd04) [inline dataRetriever/factory/containers/requestersFinder_test.go:37]:
```suggestion
func TestNewRequestersFinder(t *testing.T) {
```

File: dataRetriever/requestHandlers/requestHandler_test.go [hunk 2]
```diff
@@ -1206,21 +1205,21 @@ func TestResolverRequestHandler_RequestPeerAuthenticationsByHashes(t *testing.T)
 
 	providedHashes := [][]byte{[]byte("h1"), []byte("h2")}
 	providedShardId := uint32(15)
-	t.Run("CrossShardResolver returns error", func(t *testing.T) {
+	t.Run("CrossShardRequester returns error", func(t *testing.T) {
 		t.Parallel()
 
 		wasCalled := false
-		paResolver := &mock.PeerAuthenticationResolverStub{
+		paRequester := &dataRetrieverMocks.HashSliceRequesterStub{
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply dataRetriever/requestHandlers/requestHandler_test.go:1212]: the peer authentication requester respects the HashSliceRequester interface so it is safe to use this stub

File: factory/processing/processComponentsHandler.go [hunk 3]
```diff
@@ -88,8 +88,8 @@ func (m *managedProcessComponents) CheckSubcomponents() error {
 	if check.IfNil(m.processComponents.interceptorsContainer) {
 		return errors.ErrNilInterceptorsContainer
 	}
-	if check.IfNil(m.processComponents.resolversFinder) {
-		return errors.ErrNilResolversFinder
+	if check.IfNil(m.processComponents.requestersFinder) {
+		return errors.ErrNilRequestersFinder
 	}
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply factory/processing/processComponentsHandler.go:93]: right, added

File: update/factory/fullSyncRequestersContainerFactory.go [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+	if check.IfNil(args.Marshaller) {
+		return nil, update.ErrNilMarshalizer
+	}
+	if check.IfNil(args.ExistingRequesters) {
+		return nil, update.ErrNilRequestersContainer
+	}
+	if check.IfNil(args.PeersRatingHandler) {
+		return nil, update.ErrNilPeersRatingHandler
+	}
+
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply update/factory/fullSyncRequestersContainerFactory.go:56]: added

File: dataRetriever/factory/containers/disabled/resolversContainer.go [hunk 5]
```diff
@@ -0,0 +1,62 @@
+package disabled
+
+import (
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever/resolvers/disabled"
+)
+
+type resolversContainer struct {
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply dataRetriever/factory/containers/disabled/resolversContainer.go:8]: as discussed, we'll keep this even after cleanup for consistency on processComponents

**Review by iulianpascalau — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Pull request merged**